### PR TITLE
Refactor: Replace tracking plugin by a composable

### PR DESCRIPTION
**Pull request by antoinemetifeu**
Issue [#ACCOUNT-910](https://forge.prestashop.com/browse/ACCOUNT-910)

* Replace tracking plugin by a composable function (useSegmentTracking) and make Segment optional (for third-party storybook use)

### CODE REVIEW

File: src/composables/useSegmentTracking.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      props = args[1];
+    }
+
+    Vue.prototype.$segment?.identify(...args);
+
+    setSuperProperties(props);
+  }
+
+  function track(name, props = {}) {
+    Vue.prototype.$segment?.track(name, {...state.properties, ...props});
```
Comment (atourneriePresta) [inline src/composables/useSegmentTracking.js:39]: Doesn't pass eslint
eslint-disable-next-line no-unused-expressions

File: src/composables/useSegmentTracking.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+      superadmin: state.context.backendUser.isSuperAdmin,
+      v4_onboarded: state.context.isOnboardedV4,
+    };
+    if (args.length === 1 && typeof args[0] === 'object') {
+      props = args[0];
+    } else if (args.length >= 2 && typeof args[1] === 'object') {
+      props = args[1];
+    }
+
+    Vue.prototype.$segment?.identify(...args);
```
Comment (atourneriePresta) [inline src/composables/useSegmentTracking.js:33]: Doesn't pass eslint
eslint-disable-next-line no-unused-expressions

File: src/composables/useSegmentTracking.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  function trackPsAccountUpdateButton() {
+    track('[ACC] PSAccount Update Button Clicked');
+  }
+
+  function trackPsEventBusInstallButton() {
+    track('[ACC] PSEventBus Install Button Clicked');
+  }
+
+  function reset() {
+    Vue.prototype.$segment?.reset();
```
Comment (atourneriePresta) [inline src/composables/useSegmentTracking.js:97]: doesn't pass eslint
eslint-disable-next-line no-unused-expressions

File: src/composables/useSegmentTracking.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
+import {toRefs} from '@vue/composition-api';
+import Vue from 'vue';
```
Comment (atourneriePresta) [inline src/composables/useSegmentTracking.js:2]: Not sure if it will works in cdn with psx actually

**Comment by antoinemetifeu** [comment]
Updated w/ #117 and depends of PrestaShopCorp/ps_accounts#228

**Comment by antoinemetifeu** [comment]
Must be edited after event bus / virtualUri PRs are merged because they contains tracking changes

File: src/full.js [hunk 5]
```diff
@@ -29,18 +17,17 @@ const install = function(vue, opts = {}) {
...
     vue.use(Segment, {
-      id: contextPsAccounts.segmentApiKey,
+      id: window?.contextPsAccounts?.segmentApiKey,
       pageCategory: "ps_accounts-ui"
     });
   }
-  vue.use(Tracking, {
-    superProperties,
-  });
+  vue.us
```
Comment (GautierBoudeweel) [inline src/full.js:30]: wtf ?

**Comment by antoinemetifeu** [comment]
- Merged into #140

**Pull request closed without merging**